lib/icu.js

```diff
@@ -0,0 +1,209 @@
+        throw new TypeError('Invalid UCS2 Buffer');
+      return icu.convertToUtf8FromUcs2(source);
+    }
+  }
+};
+
+// Transcodes the Buffer from one encoding to another, returning a new
+// Buffer instance.
+exports.transcode = function transcode(source, from_enc, to_enc) {
+  if (!source || !(source.buffer instanceof ArrayBuffer))
+    throw new TypeError('"source" argument must be a Buffer');
+  if (source.length === 0) return Buffer.alloc(0);
+
+  from_enc = normalizeEncoding(from_enc);
+  to_enc = normalizeEncoding(to_enc);
+
+  if (from_enc === to_enc)
+    return Buffer.from(source);
+
+  const cnv_from = conversions[from_enc];
+
+  if (cnv_from) {
+    const cnv_to = cnv_from[to_enc];
+    if (cnv_to)
+      return cnv_to(source);
+  }
+  throw new TypeError(`Unsupported conversion: ${from_enc} to ${to_enc}`);
+};
+
+// Returns the unicode codepoint at a specific offset location in the buffer,
+// based on the given encoding
+exports.codePointAt = function(buf, pos, encoding) {
+  if (!buf || !(buf.buffer instanceof ArrayBuffer))
+    throw new TypeError('First argument must be a Buffer');
+  if (pos < 0 || pos >= buf.length)
+    return;
+  pos >>>= 0;
+
+  encoding = normalizeEncoding(encoding || 'utf8');
+
+  if (encoding === 'binary' || encoding === 'ascii')
+    return buf[pos];
+  if (encoding !== 'utf8' && encoding !== 'ucs2')
+    throw new TypeError(`Unsupported Encoding: ${encoding}`);
+
+  return icu.getCodePointAt(buf, encoding === 'utf8', pos);
+};
+
+// Returns the char at a specific offset location in the buffer, based
+// on the given encoding.
+exports.charAt = function bufferCharAt(buf, pos, encoding) {
+  if (!buf || !(buf.buffer instanceof ArrayBuffer))
+    throw new TypeError('First argument must be a Buffer');
+  if (pos < 0 || pos >= buf.length)
+    return;
+
+  pos >>>= 0;
+  encoding = normalizeEncoding(encoding || 'utf8');
+
+  if (encoding === 'binary' || encoding === 'ascii')
+    return String.fromCharCode(buf[pos]);
+  if (encoding !== 'utf8' && encoding !== 'ucs2')
+    throw new TypeError(`Unsupported Encoding: ${encoding}`);
+
+  return icu.getCharAt(buf, encoding === 'utf8', pos);
+};
+
+function utf8IsLeadByte(c) {
+  return (c & 0xc0) != 0x80;
+}
+
+function utf8CountTrailBytes(c) {
+  return ((c >= 0xc0) + (c >= 0xe0) + (c >= 0xf0));
+}
+
+function utf8ForwardN(buf, start, n) {
+  // Assumes that buf[start] is already aligned on a lead byte
+  const len = buf.length;
+  n >>>= 0;
+  var i = start;
+  if (i >= len) throw new RangeError('index out of range');
+  for (; n > 0; n--)
+    i += 1 + utf8CountTrailBytes(buf[i]);
+  return i;
+}
+
+// Performs a UTF-8 aware slice without error checking. start and end
+// represent character offsets, not byte offsets.
+function utf8Slice(buf, start, end) {
+  if (!buf || !(buf.buffer instanceof ArrayBuffer))
+    throw new TypeError('First argument must be a Buffer');
+  if (typeof end === 'undefined') end = buf.length;
+  start >>= 0;
+  end >>= 0;
```

Comment:
Yeah, I've had a standing todo to go back and reconcile this. I'll hit that in the next update.